src/mappings/staking.ts

```diff
@@ -172,26 +172,58 @@ export async function handleWithdrawClaimed(
 
     await withdrawl.save();
   } else {
-    const withdrawl = Withdrawl.create({
+    await upsertWithdrawlEntity({
       id,
       delegator: source,
       indexer: '-',
-      index: index.toBigInt(),
-      startTime: event.blockTimestamp,
-      amount: amount.toBigInt(),
+      index,
+      amount,
       claimed: true,
-      createdBlock: event.blockNumber,
+      event,
     });
 
+    logger.warn(`Force upsert: Expected withdrawl ${id} to exist.`);
+    const exception = `Expected withdrawl ${id} to exist: ${JSON.stringify(
+      event
+    )}`;
+
+    await reportException(
+      'handleWithdrawClaimed',
+      event.logIndex,
+      event.blockNumber,
+      exception
+    );
+  }
+}
+
+export async function handleWithdrawCancelled(
+  event: AcalaEvmEvent<UnbondCancelledEvent['args']>
+): Promise<void> {
+  const { source, indexer, amount, index } = event.args;
+  const id = getWithdrawlId(source, index);
+  const withdrawl = await Withdrawl.get(id);
+
+  if (withdrawl) {
+    withdrawl.lastEvent = `handleWithdrawCancelled:${event.blockNumber}`;
     await withdrawl.save();
+  } else {
+    await upsertWithdrawlEntity({
+      id,
+      delegator: source,
+      indexer,
+      index,
+      amount,
+      claimed: false,
+      event,
+    });
 
```

Comment:
Unless you want remove the entity handleWithdrawCancelled, but this has historically been a bad idea